File: C/BubbleSort.c
// Bubble Sort
#include <stdio.h>

int main()
{
    int size;
    printf("Enter the size of the array: ");

    // Validate input for size
    if (scanf("%d", &size) != 1 || size <= 0) {
        printf("Invalid input. Please enter a positive integer for the size of the array.\n");
        return 1; // Exit the program with an error code
    }
    
    int arr[size];
    input(arr,size);
    sortArray(arr,size);
    display(arr,size);
}

void input(int b[],int N)
{
    printf("Enter values of %d array: ",N);
    for(int i = 0;i < N; i++)
    {
        scanf("%d",&b[i]);
    }
}

void display(int b[],int N)
{
    printf("Enter values of array are: ");
    for(int i = 0; i < N; i++)
    {
        printf("%d ",b[i]);
    }
    printf("\n");
}

void sortArray(int b[],int N)
{
    int temp;
    for(int i = 0; i < N; i++)
    {
        for(int j = 0; j < N-i; j++)
        {
            if(b[j] > b[j+1])
            {
                temp = b[j+1];
                b[j+1] = b[j];
                b[j] = temp;
            }
        }
    }
}

File: Python/BubbleSort.py
def input_array(N):
    try:
        arr = []
        print(f"Enter values of {N} array:")
        for i in range(N):
            arr.append(int(input()))
        return arr
    except ValueError:
        print("Invalid input. Please enter integers.")
        exit(1)

def display_array(arr):
    print("Values of the array are:", end=" ")
    for element in arr:
        print(element, end=" ")
    print()

def sort_array(arr):
    n = len(arr)
    for i in range(n):
        for j in range(0, n-i-1):
            if arr[j] > arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]

def main():
    try:
        size = int(input("Enter the size of the array: "))
        if size <= 0:
            print("Invalid input. Please enter a positive integer for the size of the array.")
            exit(1)

        arr = input_array(size)
        sort_array(arr)
        display_array(arr)
    except KeyboardInterrupt:
        print("\nProgram terminated by user.")
        exit(1)

if __name__ == "__main__":
    main()
